sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -1072,20 +1072,22 @@ def CreateItem(self, database_or_container_link, document, options=None, **kwarg
         # not each time the function is called (like it is in say, Ruby). This means
         # that if you use a mutable default argument and mutate it, you will and have
         # mutated that object for all future calls to the function as well. So, using
-        # a non-mutable deafult in this case(None) and assigning an empty dict(mutable)
+        # a non-mutable default in this case(None) and assigning an empty dict(mutable)
         # inside the method For more details on this gotcha, please refer
         # http://docs.python-guide.org/en/latest/writing/gotchas/
         if options is None:
             options = {}
 
         # We check the link to be document collection link since it can be database
         # link in case of client side partitioning
-        if base.IsItemContainerLink(database_or_container_link):
-            options = self._AddPartitionKey(database_or_container_link, document, options)
-
         collection_id, document, path = self._GetContainerIdWithPathForItem(
             database_or_container_link, document, options
         )
+
+        if base.IsItemContainerLink(database_or_container_link):
+            print('checking link', document)
```

Comment:
A print statement ;)
Additionally - why was this block moved down? Does the ordering change here affect the `options`?

Reply:
Oops, I thought that print statement was already there when I reordered them. Let me remove! The order changed because the ID is generated and added to the body in the statement above this one. This statement then adds the `partitionKey` to the options. In the case that `partitionKey` is `/id`, we won't have the `partitionKey` in the options unless the statements are in this order

Reply:
Fixed